apps/server/src/shared/infra/identity-management/keycloak-identity-management.spec.ts

```diff
@@ -0,0 +1,255 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import KeycloakAdminClient from '@keycloak/keycloak-admin-client';
+import { IdentityManagement } from './identity-management';
+import { KeycloakIdentityManagement } from './keycloak-identity-management';
+
+describe('KeycloakIdentityManagement', () => {
+	let idm: IdentityManagement;
+	let kcAdminClient: KeycloakAdminClient;
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [
+				{ provide: 'IdentityManagement', useClass: KeycloakIdentityManagement },
+				{
+					provide: 'KeycloakAdminClient',
+					useValue: {
```

Comment:
Can we use a `createMock<KeycloakAdminClient>` mock factory here? e.g. 
https://github.com/hpi-schul-cloud/schulcloud-server/blob/25e27a21ca32106008f56aba8fc1973285e4282e/apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts#L59

Reply:
@uidp We tried that first, yet createMock does not seem to support deep mocking for classes but functions only. Hence internals, e.g. 'kcAdminClient.users' are not mocked. Do you know an alternative?

Reply:
Yes, the properties are not mocked, only the  methods. But I guess the properties can just be assigned to give them a value, can't they?

Reply:
@uidp As mentioned below, these properties are classes and we need to mock their internals as well. I'll stick with the jest vanilla approach.